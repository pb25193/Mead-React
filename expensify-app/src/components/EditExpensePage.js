import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';



export class EditExpensePage extends React.Component {

    handleEditButton = (updates)=>{
        this.props.startEditExpense(this.props.expense.id, updates);
        this.props.history.push('/dashboard');
    }

    handleRemoveButton = ()=>{
        this.props.startRemoveExpense(this.props.expense.id);
        this.props.history.push('/dashboard');
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        expense={this.props.expense}
                        onSubmit={this.handleEditButton}
                    />
                    <button 
                        className="button--danger button"
                        onClick = {this.handleRemoveButton} 
                    >Remove Expense</button>
                </div>
            </div>
        );
    }

}


const matchDispatchToProps = (dispatch) => ({
    startEditExpense: (id, updates) => dispatch(startEditExpense({id, updates})),
    startRemoveExpense: (id) => dispatch(startRemoveExpense({id}))
});

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense)=> expense.id === props.match.params.id)
});

export default connect(mapStateToProps, matchDispatchToProps)(EditExpensePage);