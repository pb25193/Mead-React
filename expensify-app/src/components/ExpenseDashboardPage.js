import React from 'react';



const ExpenseDashboardPage = () => (
    <div>
        ExpenseDashBoardPage
    </div>
);

export default ExpenseDashboardPage;