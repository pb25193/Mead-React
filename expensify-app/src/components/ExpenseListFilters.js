import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { editFilterText, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'
import 'react-dates/lib/css/_datepicker.css';


class ExpenseListFilters extends React.Component{

    state = {
        calendarFocused: null
    }

    onDatesChange = ({ startDate, endDate }) => {
        console.log(startDate, endDate);
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }

    onFocusChange = (calendarFocused) => {
        this.setState(()=>({ calendarFocused }));
    }

    render(){
        return (
            <div>
                <input 
                    type='text' 
                    value={this.props.filters.text} 
                    onChange = {(e) => this.props.dispatch(editFilterText(e.target.value))} 
                />
                <select
                    value={this.props.filters.sortBy}
                    onChange = {(e)=>{
                        if(e.target.value === "amount") this.props.dispatch(sortByAmount());
                        else if(e.target.value === "date") this.props.dispatch(sortByDate());
                    }}
                >
                    <option value = "date">Date</option>
                    <option value = "amount">Amount</option>
                </select>
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={(this.onFocusChange)}
                    showClearDates={true}
                />
            </div>
        );
    }
};


const mapStateToProps = (state) => ({
    filters: state.filters
});


export default connect(mapStateToProps)(ExpenseListFilters);

