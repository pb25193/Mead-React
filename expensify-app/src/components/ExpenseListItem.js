import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}:</h3> 
        </Link>
        <div>{amount}, created at {createdAt}</div> 
    </div>
);


export default connect()(ExpenseListItem);

