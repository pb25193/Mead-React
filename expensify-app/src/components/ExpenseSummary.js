import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import numeral from 'numeral';
import { getVisibleExpenses } from '../selectors/expenses';
import totalExpenses from '../selectors/expense-total';

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => (   
    <div className="page-header">
        <div className="content-container">
            <h1
                className="page-header__title"
            >
                Viewing <span>{expenseCount}</span> expense{expenseCount===1 ? "" : "s"} totalling <span>{numeral(expensesTotal/100).format('$0,0.00')}</span>
            </h1>
            <div className="page-header__actions">
                <Link className="button" to="/create">Add Expense</Link>
            </div>
        </div>
    </div>
);

const mapStateToProps = (state) => {
    const expenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        expenseCount: expenses.length,
        expensesTotal: totalExpenses(expenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);