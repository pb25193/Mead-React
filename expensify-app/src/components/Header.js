import React from 'react';
import { NavLink } from 'react-router-dom';





const Header = () => (
    <div>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active" exact={true}>Add Expense</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit</NavLink>
    </div>
);


export default Header;