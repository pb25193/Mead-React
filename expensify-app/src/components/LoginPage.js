import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

const LoginPage = ({ startLogin }) => (
    <div>
        <button
            onClick={startLogin}
        >Log in</button>
    </div>
);

const mapDispatchToProps = (dispatch)=>({
    startLogin: () => dispatch(startLogin())
});

const ConnectedLoginPage = connect(undefined, mapDispatchToProps)(LoginPage);

export { LoginPage, ConnectedLoginPage as default };