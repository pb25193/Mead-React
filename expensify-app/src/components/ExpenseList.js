import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import { getVisibleExpenses } from '../selectors/expenses'


const ExpenseList = (props) => (
    <div>
        <h1>Expense List for </h1>
        {props.expenses.map((item) => (<ExpenseListItem 
            key={item.id} 
            {...item}
        />))}
    </div>
);

const mapStateToProps = (state) => ({
    expenses: getVisibleExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);

