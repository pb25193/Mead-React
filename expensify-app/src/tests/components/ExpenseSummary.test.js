import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should render correctly with 1 expense', ()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render correctly with many expenses', ()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expensesTotal={23213443255} />);
    expect(wrapper).toMatchSnapshot();
});