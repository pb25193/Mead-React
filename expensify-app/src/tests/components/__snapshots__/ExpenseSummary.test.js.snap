// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       expense
       totalling 
      <span>
        $2.35
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render correctly with many expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       expense
      s
       totalling 
      <span>
        $232,134,432.55
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
